miden-crypto/src/word.rs

```diff
@@ -0,0 +1,616 @@
+    pub const fn new(value: [Felt; WORD_SIZE_FELT]) -> Self {
+        Self(value)
+    }
+
+    /// Returns the digest as a slice of field elements.
+    pub fn as_elements(&self) -> &[Felt] {
+        self.as_ref()
+    }
+
+    /// Returns the digest as a byte array.
+    pub fn as_bytes(&self) -> [u8; WORD_SIZE_BYTES] {
+        <Self as Digest>::as_bytes(self)
+    }
+
+    /// Returns an iterator over the elements of multiple digests.
+    pub fn digests_as_elements_iter<'a, I>(digests: I) -> impl Iterator<Item = &'a Felt>
+    where
+        I: Iterator<Item = &'a Self>,
+    {
+        digests.flat_map(|d| d.0.iter())
+    }
+
+    /// Returns all elements of multiple digests as a slice.
+    pub fn digests_as_elements(digests: &[Self]) -> &[Felt] {
+        let p = digests.as_ptr();
+        let len = digests.len() * WORD_SIZE_FELT;
+        unsafe { slice::from_raw_parts(p as *const Felt, len) }
+    }
+
+    /// Returns hexadecimal representation of this digest prefixed with `0x`.
+    pub fn to_hex(&self) -> String {
+        bytes_to_hex_string(self.as_bytes())
+    }
+}
+
+impl Hash for Word {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(&self.as_bytes());
+    }
+}
+
+impl Digest for Word {
+    fn as_bytes(&self) -> [u8; WORD_SIZE_BYTES] {
+        let mut result = [0; WORD_SIZE_BYTES];
+
+        result[..8].copy_from_slice(&self.0[0].as_int().to_le_bytes());
+        result[8..16].copy_from_slice(&self.0[1].as_int().to_le_bytes());
+        result[16..24].copy_from_slice(&self.0[2].as_int().to_le_bytes());
+        result[24..].copy_from_slice(&self.0[3].as_int().to_le_bytes());
+
+        result
+    }
+}
+
+impl Deref for Word {
+    type Target = [Felt; WORD_SIZE_FELT];
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Ord for Word {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Compare the inner u64 of both elements.
+        //
+        // It will iterate the elements and will return the first computation different than
+        // `Equal`. Otherwise, the ordering is equal.
+        //
+        // Finally, we use `Felt::inner` instead of `Felt::as_int` so we avoid performing a
+        // montgomery reduction for every limb. that is safe because every inner element of the
+        // digest is guaranteed to be in its canonical form (that is, `x in [0,p)`).
+        self.0.iter().map(Felt::inner).zip(other.0.iter().map(Felt::inner)).fold(
+            Ordering::Equal,
+            |ord, (a, b)| match ord {
+                Ordering::Equal => a.cmp(&b),
+                _ => ord,
+            },
+        )
+    }
+}
+
+impl PartialOrd for Word {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Display for Word {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let encoded: String = self.into();
+        write!(f, "{encoded}")?;
+        Ok(())
+    }
+}
```

Comment:
```suggestion
impl Display for Word {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        write!(f, "{}", self.to_hex())
    }
}
```

Nit: Not your code of course, but I feel like this would be much easier to read and communicate more clearly that the string representation is hex.